packages/engine/src/scene/components/VariantComponent.tsx

```diff
@@ -123,8 +124,18 @@ export const VariantComponent = defineComponent({
   reactor: () => {
     const entity = useEntityContext()
     const variantComponent = useComponent(entity, VariantComponent)
-
     const instancingComponent = useOptionalComponent(entity, InstancingComponent)
+    let childEntity = getChildrenWithComponents(entity, [GLTFComponent]).at(0)
+
+    useEffect(() => {
+      childEntity = createEntity()
```

Comment:
this useEffect should be dependent on not having an instancing component, since it handles loading models separately, and needs to remove the entity on it's unmount. essentially, this useEffect can be merged with the next one, and the childEntity can be scoped entirely to the next useEffect, cleaned up with removeEntity in the return function